java/vector/src/main/java/org/apache/arrow/vector/LargeVarBinaryVector.java

```diff
@@ -112,6 +113,20 @@ public byte[] get(int index) {
     return result;
   }
 
+  /**
+   * Read the value at the given position to the given output buffer.
+   * The caller is responsible for checking for nullity first.
+   *
+   * @param index position of element.
+   * @param outputBuffer the buffer to write into.
+   */
+  public void readBytes(int index, ReusableBuffer<?> outputBuffer) {
+    final long startOffset = getStartOffset(index);
+    final int dataLength =
```

Comment:
Use getEndOffset instead? Do we need to worry about lengths exceeding max integer?

Reply:
Oddly, getEndOffset() is part of BaseVariableWidthVector and _not_ part of BaseLargeVariableWidthVector, and BLVW doesn't extend VLW.

The existing getters on this class do the narrowing cast from long to int as well, so if this is a problem the scope is larger than this.

Reply:
And I think that it's not possible for the current implementation to exceed `int`. However a Java 21+ memory module should actually be able to.

@danepitkin we should probably file some follow-up work for after your PR lands to shake out all the problems that are about to crop up.